pkg/common/providers/ocmprovider/cluster.go

```diff
@@ -390,16 +391,49 @@ func (o *OCMProvider) GenerateProperties() (map[string]string, error) {
 
 // DeleteCluster requests the deletion of clusterID.
 func (o *OCMProvider) DeleteCluster(clusterID string) error {
-	var resp *v1.ClusterDeleteResponse
+	var deleteResp *v1.ClusterDeleteResponse
+	var resumeResp *v1.ClusterResumeResponse
 	var cluster *spi.Cluster
 	var err error
-	var ok bool
 
-	if cluster, ok = o.clusterCache[clusterID]; !ok {
-		cluster, err = o.GetCluster(clusterID)
-		if err != nil {
-			return fmt.Errorf("error retrieving cluster for deletion: %v", err)
+	cluster, err = o.GetCluster(clusterID)
+	if err != nil {
+		return fmt.Errorf("error retrieving cluster for deletion: %v", err)
+	}
+
+	if cluster.State() == spi.ClusterStateHibernating || cluster.State() == spi.ClusterStateResuming {
+		if cluster.State() == spi.ClusterStateHibernating {
+			if err = retryer().Do(func() error {
+				var err error
+				resumeResp, err = o.conn.ClustersMgmt().V1().Clusters().Cluster(clusterID).Resume().Send()
+
+				if err != nil {
+					return fmt.Errorf("couldn't resume cluster '%s': %v", clusterID, err)
+				}
+
+				if resumeResp != nil && resumeResp.Error() != nil {
+					err = errResp(resumeResp.Error())
+					log.Printf("%v", err)
+					return err
+				}
+
+				return nil
+			}); err != nil {
+				return err
+			}
 		}
+
+		wait.PollImmediate(1*time.Minute, 15*time.Minute, func() (bool, error) {
+			cluster, err = o.GetCluster(clusterID)
+			if err != nil {
+				log.Printf("error retrieving cluster for deletion: %v", err)
+				return false, nil
+			}
+			if cluster.State() == spi.ClusterStateReady {
+				return true, nil
+			}
+			return false, nil
+		})
```

Comment:
I think that writing it in this (untested) order reads a little more easily, but I also kinda suspect that I'm missing a nuance here that makes this a bad idea:
```suggestion
	if cluster.State() == spi.ClusterStateHibernating {
		if err = retryer().Do(func() error {
			var err error
			resumeResp, err = o.conn.ClustersMgmt().V1().Clusters().Cluster(clusterID).Resume().Send()

			if err != nil {
				return fmt.Errorf("couldn't resume cluster '%s': %v", clusterID, err)
			}

			if resumeResp != nil && resumeResp.Error() != nil {
				err = errResp(resumeResp.Error())
				log.Printf("%v", err)
				return err
			}

			return nil
		}); err != nil {
			return err
		}
	}
// hopefully the cluster state changes by now
	if cluster.State() == spi.ClusterStateResuming {

		wait.PollImmediate(1*time.Minute, 15*time.Minute, func() (bool, error) {
			cluster, err = o.GetCluster(clusterID)
			if err != nil {
				log.Printf("error retrieving cluster for deletion: %v", err)
				return false, nil
			}
			if cluster.State() == spi.ClusterStateReady {
				return true, nil
			}
			return false, nil
		})
```

Reply:
I took a different approach simplifying it, mind taking a look again? :)